SlimeRancher.asl

```diff
@@ -1,54 +1,71 @@
+    int keys        : "UnityPlayer.dll", 0x14D4228, 0x8, 0x8, 0x228, 0x28, 0x30, 0x90;
+    double worldTime: "UnityPlayer.dll", 0x14D4228, 0x8, 0x8, 0x218, 0x28, 0x60, 0x50;
 }
 
 state("SlimeRancher", "1.3.0b") {
-  bool creditsFlag : "SlimeRancher.exe", 0x14E4F38, 0x10, 0x0, 0x20, 0x8, 0x10, 0x28, 0x58, 0x49;
-  //int gameState : "SlimeRancher.exe", 0x141D330; unused as of now
-  //int keys : ""; add support for this later
-  double worldTime : "SlimeRancher.exe", 0x14C9910, 0x8, 0xF8, 0x28, 0x48, 0xA8, 0x70;
+    bool creditsFlag: 0x14C9910, 0x0, 0xB8, 0x28, 0x198, 0x150, 0x48;
+    //int gameState : 0x141D330;
+    //int keys      : ""; requires sig scan
+    double worldTime: 0x14C9910, 0x8, 0xF8, 0x28, 0x48, 0xA8, 0x70;
 }
 
 startup {
-  // -- SETTINGS --
-  settings.Add("key_get", false, "Split upon grabbing a key");
+    settings.Add("keysplits", false, "1.4.0b ONLY: Split when collecting a new key in a run (non-specific)");
+        settings.Add("key1", false, "Key N° 1", "keysplits");
+        settings.Add("key2", false, "Key N° 2", "keysplits");
+        settings.Add("key3", false, "Key N° 3", "keysplits");
+        settings.Add("key4", false, "Key N° 4", "keysplits");
+        settings.Add("key5", false, "Key N° 5", "keysplits");
+        settings.Add("key6", false, "Key N° 6", "keysplits");
+        settings.Add("key7", false, "Key N° 7", "keysplits");
+        settings.Add("key8", false, "Key N° 8", "keysplits");
+        settings.Add("key9", false, "Key N° 9", "keysplits");
+        settings.Add("key10", false, "Key N° 10", "keysplits");
+        settings.Add("key11", false, "Key N° 11", "keysplits");
+
+    settings.Add("gatesplits", false, "1.4.0b ONLY: Split when using a key on a gate");
 }
 
 init {
-  //--Determine game version--// 
-  switch(modules.First().ModuleMemorySize) {
-    case 0x1718000:
-      version = "1.3.0b";
-      break;
-    case 0xA4000:
-      version = "1.4.0b";
-      break;
-    default:
-      print("Game version not detected or not supported");
-      break;
-  }
+    switch(modules.First().ModuleMemorySize) {
+        case 0x1718000:
+            version = "1.3.0b";
+            break;
+        case 0xA4000:
+            version = "1.4.0b";
+            break;
+        default:
+            print(">>>>> Game version not detected or not supported.");
+            break;
+    }
+
+    vars.keysInRun = 0;
+}
+
+update {
+    if (old.keys < current.keys && version == "1.4.0b") {
+        vars.keysInRun++;
+        print(">>>> keysInRun has increased to " + vars.keysInRun);
+    }
 }
 
 start {
-  return current.worldTime < 32460 //prevent starting on non-new files
-  && old.worldTime < current.worldTime //make sure time is moving
-  && current.worldTime >= 32402; //check if time has moved  (set to roughly .03 seconds after initial world time on new file)
+    if (old.worldTime < current.worldTime && current.worldTime >= 32401 && current.worldTime < 32410) {
```

Comment:
There are issues on slower machines where half the time, the timer starts before the introduction text ends.  Reducing the time check from 32402 to 32401 will likely make this happen more often even though in theory it makes the timer more accurate.

Reply:
I was aware of the world timer starting in the "intro" section, that's why I kept the number above 32401. However, this start never exceeded 32400.49 for me. I do believe you however, when you say that on slower machines it may differ. I've updated my changes, please take another look.